Fix for #218

cdancy opened the pull request:
6 failures running integration tests locally but they do not appear to be related to this change. All look to be coming from commands being run inside containers and returning non-zero exit codes.

cdancy commented:
Well it passed here so I guess that's good ;)

cdancy commented on src/main/java/com/github/dockerjava/api/model/LogConfig.java:
```diff
@@ -0,0 +1,36 @@
+    public String type = "json-file";
+    
+    @JsonProperty("Config")
+    public List<String> config = null;
```
It should be noted this variable does nothing through docker at the moment (and I vaguely remember talk of it being removed) and serves only to help tests run successfully.

marcuslinke replied:
According to the docs `config` must be a `Map<String, String>`

cdancy replied:
Fixed.

marcuslinke commented on src/main/java/com/github/dockerjava/api/model/LogConfig.java:
```diff
@@ -0,0 +1,36 @@
+public class LogConfig {
+    
+    @JsonProperty("Type")
+    public String type = "json-file";
```
There shouldn't be a default value.

cdancy replied:
Gotcha. Removed.

marcuslinke commented:
It would be the best to have at least a test case for LogConfig in `CreateContainerCmdImplTest`. Any chance?

cdancy commented:
That's the thing: no matter what one puts in for values the command to create a container succeeds. If I put in "fake-log-driver" for type docker will not complain and just default to log-driver none. If I put in configs that have nothing to do with the type docker will just ignore them. 

So with that in mind I'm not sure what we could even test. I've made the requested changes and fleshed out the support a bit more for this option.

marcuslinke commented:
I think a basic test could create a container with a valid log config (like `json-file`) and inspect the container afterwards to check if the settings are accepted by the docker host.

marcuslinke commented:
Don't worry about failed intergration tests. CircleCI uses a quite old docker version where log config is not supported.

cdancy commented:
I've added the test 'CreateContainerCmdImplTest.createContainerWithLogConfig' which sets LogConfig to 'none' and inspects the reponse to ensure that is what is actually set. Works here against my local docker instance. I added the ' @Test(groups = "ignoreInCircleCi")' annotation as I assume we don't want this run on CircleCI due to it using an older docker version.

marcuslinke commented:
@cdancy I've modified your code so the logging type is an enum now. Thanks for your contribution!

cdancy commented:
Good call. I was thinking of doing the same but wasn't sure if we should force the user to pass in the Enum on the LogConfig constructor/setters or do a check to ensure the type was supported.
